database_client/models.py

```diff
@@ -137,6 +137,51 @@ class Base(DeclarativeBase):
         str_255: String(255),
     }
 
+    @hybrid_method
+    def to_dict(cls, subquery_parameters=[]):
+        # Calls the class's __iter__ implementation
+        dict_result = dict(cls)
+        keyorder = cls.__mapper__.column_attrs.items()
+
+        for param in subquery_parameters:
+            if param.linking_column not in dict_result:
+                dict_result[param.linking_column] = []
+
+        sorted_dict = {
+            col: dict_result[col] for col, descriptor in keyorder if col in dict_result
+        }
+        sorted_dict.update(dict_result)
+
+        return sorted_dict
+
+
+class CountMetadata:
+    @hybrid_method
```

Comment:
Breaking up the metadata components into separate classes is a great idea. In the long term, I'll want to think about the degree we're applying inheritance in some of these cases, but for now, this helps keep the code footprint low.